### PR TITLE
416 Adds DRP exercise endpoint

### LINKED ISSUE #416: [DRP] Add support for POST /exercise endpoint
### Requirements

1. MUST expose a POST /api/v1/drp/exercise endpoint (similar to POST /api/v1/privacy-request)
3. MUST support all required request body fields as defined by the [specification](https://github.com/consumer-reports-digital-lab/data-rights-protocol#202-post-exercise-data-rights-exercise-endpoint) (meta, exercise, identity); note that regime, relationships, and status_callback are covered by other requirements below
5. MUST lookup the Fidesops policy to use for the request based on the “DRP action” (instead of based on a Fidesops policy_key)
7. SHOULD throw an error if multiple “DRP actions” are specified in the exercise field; Fidesops requests currently only support a single policy per request
9. MUST decode the DRP identity JWT token and validate the included identity fields based on the [specification](https://github.com/consumer-reports-digital-lab/data-rights-protocol#304-schema-identity-encapsulation) (name, email, phone_number, address)
11. MUST map all valid Fidesops request status to DRP “Request Statuses” as defined in the [specification](https://github.com/consumer-reports-digital-lab/data-rights-protocol#303-schema-status-of-a-data-subject-exercise-request) (e.g. open, in_progress, fulfilled)
13. MUST create the privacy request otherwise normally, using the given DRP identity and DRP action 
15. MUST return a JSON response matching the “Exercise Status” schema defined in the DRP [specification](https://github.com/consumer-reports-digital-lab/data-rights-protocol#303-schema-status-of-a-data-subject-exercise-request) (request_id, status, received_at, etc.)
17. SHOULD support the status_callback 
19. SHOULD cache the contents of the request body in the identity cache (for troubleshooting)

### PR DESCRIPTION
**Pull request by eastandwestwind**
# Purpose
Adds DRP exercise endpoint

# Changes
- Adds endpoint for DRP exercise
- Refactors DRP status endpoint into DRP-specific endpoint file
- Adds unit tests for DRP exercise
- Adds DRP folder to postman collection 
- Additionally, removes ignore_missing_imports flag for mypy

# Checklist

- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
- [x] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #416

### CODE REVIEW

File: src/fidesops/models/privacy_request.py
Comment (eastandwestwind) [inline src/fidesops/models/privacy_request.py:190]: cache doesn't handle vals as bytes/str/int/float. Perhaps we should instead cache the whole req body as a str (`json.loads()`) instead of separating the top level props as cache keys. But this works as far as caching for debugging purposes only.

File: src/fidesops/schemas/drp_privacy_request.py
Comment (eastandwestwind) [inline src/fidesops/schemas/drp_privacy_request.py:27]: DRP spec dictates that this is a list. So we'll take in a list even though we don't support > 1 item

**Comment by eastandwestwind** [comment]
Verified both endpoints manually in postman:

Screen Shot 2022-05-11 at 2 29 41 PM

Screen Shot 2022-05-11 at 2 27 03 PM

File: src/fidesops/core/config.py
Comment (eastandwestwind) [inline src/fidesops/core/config.py:123]: should I also add this to `fidesops.toml` as an empty str?

File: src/fidesops/api/v1/endpoints/drp_endpoints.py
Comment (seanpreston) [inline src/fidesops/api/v1/endpoints/drp_endpoints.py:60]: This isn't really a `400` since if this variable isn't in the config it represents a server misconfiguration. `501_NOT_IMPLEMENTED` or `500_SERVER_ERROR` might be best.

File: src/fidesops/api/v1/endpoints/drp_endpoints.py
Comment (seanpreston) [inline src/fidesops/api/v1/endpoints/drp_endpoints.py:115]: These `except` blocks should surround only the minimal code, and we should avoid using broad excepts unless there's a specific reason?

File: src/fidesops/api/v1/endpoints/drp_endpoints.py
Comment (seanpreston) [inline src/fidesops/api/v1/endpoints/drp_endpoints.py:154]: Thanks for moving this into this file!

File: src/fidesops/api/v1/urn_registry.py
Comment (seanpreston) [inline src/fidesops/api/v1/urn_registry.py:94]: NIT: can we store URLs as absolute strings so that they're easily `cmd + f`-able?

File: src/fidesops/core/config.py
Comment (seanpreston) [reply src/fidesops/core/config.py:123]: Yes please!

File: src/fidesops/service/drp/drp_fidesops_mapper.py
Comment (seanpreston) [inline src/fidesops/service/drp/drp_fidesops_mapper.py:9]: Nice implementation 🙌

File: src/fidesops/service/drp/drp_fidesops_mapper.py
Comment (seanpreston) [inline src/fidesops/service/drp/drp_fidesops_mapper.py:25]: Do we want to discard any extra attributes on `DrpIdentity` silently?

File: tests/api/v1/endpoints/test_drp_endpoints.py
Comment (seanpreston) [inline tests/api/v1/endpoints/test_drp_endpoints.py:47]: It's probably worth setting a default for this value that's overridden on deployment [in the same way we do with the root client](https://github.com/ethyca/fidesops/blob/main/fidesops.toml#L20-L21). That way we won't need to override it here?

File: src/fidesops/api/v1/endpoints/drp_endpoints.py
Comment (eastandwestwind) [reply src/fidesops/api/v1/endpoints/drp_endpoints.py:60]: ah right, good point

File: src/fidesops/api/v1/endpoints/drp_endpoints.py
Comment (eastandwestwind) [reply src/fidesops/api/v1/endpoints/drp_endpoints.py:115]: this was to be consistent with what we have for POST `"/privacy-request"`. But I can remove it, just lemme know your preference

File: src/fidesops/api/v1/urn_registry.py
Comment (eastandwestwind) [reply src/fidesops/api/v1/urn_registry.py:94]: should we eventually update the rest of the `urn_registry` to be `cmd + f`-able in the same way?

File: src/fidesops/models/privacy_request.py
Comment (seanpreston) [inline src/fidesops/models/privacy_request.py:193]: I know this probably feels hacky but good choice in effort/reward trade-off if this is only for debug purposes as you say.

Looks like `value` can only ever be a key from [this schema](https://github.com/ethyca/fidesops/blob/579f028faa620b04c34e58187c1bb212f62bf535/src/fidesops/schemas/drp_privacy_request.py#L22), which should be just fine.

If we want to do more with this data though let's make sure we're breaking it down and storing it by key. We could write a util method to handle that automatically for Pydantic schemas for instance.

**Review by seanpreston — APPROVED** [review]
LGTM — would be nice if we could address the URL nit and JWT config var before merging!

File: src/fidesops/service/drp/drp_fidesops_mapper.py
Comment (eastandwestwind) [reply src/fidesops/service/drp/drp_fidesops_mapper.py:25]: I'll surface a warning log message using a validator or a check within this mapper

**Comment by eastandwestwind** [comment]
Thanks for the review @seanpreston! This is good to go.

I've added a warning message in case we don't support the specific identity props passed in: https://github.com/ethyca/fidesops/pull/496/files#diff-e2fc0ccbfb0913f3f97fd1f425377e29beb158ea9581cc5e171e044b8927c4a4R30

**Pull request merged**
